src/pytorch_lightning/utilities/data.py

```diff
@@ -371,18 +432,30 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
         )
         param_names = param_names[: len(args)]
 
-        if not hasattr(obj, "__pl_dl_args"):
-            obj.__pl_dl_args = args
-            obj.__pl_dl_kwargs = kwargs
-            obj.__pl_dl_arg_names = param_names
+        default_kwargs = {
+            param.name: param.default
+            for param in params.values()
+            if param.name != "self"
+            and param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)
```

Comment:
This is very similar to the tuple created in L428. Could we create this dict first and extract the tuple from the keys of this dict?

Then you could remove the filter `(param.name not in kwargs and param.name not in param_names)`

Reply:
So, you're right about the fact, that it is very similar. I changed it according to your suggestion and honestly, it doesn't like like such an improvement to me. The only advantage I see is that now the intention of both of these structures is a little bit clearer.

Reply:
I like this version better. Thanks!